yaml_edit/lib/src/editor.dart

```diff
@@ -613,10 +589,8 @@ class YamlEditor {
   /// the whole tree.
   ///
   /// [SourceSpan]s in this new tree are not guaranteed to be accurate.
-  YamlNode _deepModify(YamlNode tree, Iterable<Object> path,
-      Iterable<Object> subPath, YamlNode expectedNode) {
-    ArgumentError.checkNotNull(path, 'path');
-    ArgumentError.checkNotNull(tree, 'tree');
+  YamlNode? _deepModify(YamlNode tree, Iterable<Object?> path,
+      Iterable<Object?> subPath, YamlNode? expectedNode) {
```

Comment:
```suggestion
      Iterable<Object?> subPath, YamlNode expectedNode) {
```

Shouldn't callers give a `YamlNode` with `.value == null` ?

Reply:
Will include this in the next set of changes.